src/python/pants/init/extension_loader.py

```diff
@@ -130,6 +130,34 @@ def load_backends_and_plugins(
             deprecation_start_version="1.28.0.dev0",
             hint="\n".join(msg_lines),
         )
+
+    if "pants.contrib.awslambda.python" in backends1:
+        msg_lines = [
+            fill(
+                (
+                    "The original awslambda implementation is being replaced by an improved "
+                    "implementation made possible by the V2 engine."
+                ),
+                80,
+            ),
+            "",
+            fill(
+                (
+                    "To prepare for this change, remove 'pants.contrib.awslambda.python' from the "
+                    "`backend_packages` entry in the `GLOBAL` section in your `pants.toml` "
+                    "(or `pants.ini`). Then add `pants.backend.awslambda.python` to the "
+                    "`backend_packages2` entry. Ensure that you have `--v2` enabled (the default)."
```

Comment:
There's more to it, isn't there? In V1, you specify the `binary` field. In V2, that field doesn't exist and you instead declare `sources` and `dependencies`.

Iirc, Asher had one more issue with migrating from V1 to V2 in the Toolchain repo. I can't remember what it was, maybe this.

Reply:
Oh, yeah, good point.